src/test/java/htsjdk/samtools/util/SamLocusIteratorTest.java

```diff
@@ -582,6 +658,10 @@ public void testOverlappingGappedAlignmentsWithoutIndels() {
             Assert.assertEquals(li.size(), expectedDepths[i]);
             Assert.assertEquals(li.getPosition(), expectedReferencePositions[i]);
             Assert.assertEquals(li.getRecordAndOffsets().size(), expectedReadOffsets[i].length);
+            // Check the correct assignment of the alignment type
+            for (final SamLocusIterator.RecordAndOffset rao : li.getRecordAndOffsets()) {
```

Comment:
This is all a bit redundant, but it just seems like these tests have a ton of redundant code.  We can clean that up separately.